src/Compilers/CSharp/Portable/Symbols/ReferenceManager.cs

```diff
@@ -986,18 +984,10 @@ internal MetadataImportOptions EffectiveImportOptions
 
                 protected override void AddAvailableSymbols(List<AssemblySymbol> assemblies)
                 {
-                    // accessing cached symbols requires a lock
-                    lock (SymbolCacheAndReferenceManagerStateGuard)
-                    {
-                        foreach (var assembly in CachedSymbols)
-                        {
-                            var peAssembly = assembly as PEAssemblySymbol;
-                            if (IsMatchingAssembly(peAssembly))
-                            {
-                                assemblies.Add(peAssembly);
-                            }
-                        }
-                    }
+                    CachedSymbols.CopyTo<AssemblySymbol, AssemblyDataForFile>(
```

Comment:
>AssemblySymbol [](start = 41, length = 14)

```PEAssemblySymbol```?

Reply:
Can't be `PEAssemblySymbol` as the collection is `List<AssemblySymbol>`. The choice here is the perf of a cast or the complexity of yet another generic parameter.